sdks/python/container/boot.go

```diff
@@ -105,18 +107,57 @@ func main() {
 
 	// (2) Retrieve and install the staged packages.
 
-	dir := filepath.Join(*semiPersistDir, "staged")
+	func() {
 
-	files, err := artifact.Materialize(ctx, *artifactEndpoint, info.GetRetrievalToken(), dir)
-	if err != nil {
-		log.Fatalf("Failed to retrieve staged files: %v", err)
-	}
+        installCompleteFile := filepath.Join(os.TempDir(), "beam.install.complete")
 
-	// TODO(herohde): the packages to install should be specified explicitly. It
-	// would also be possible to install the SDK in the Dockerfile.
-	if setupErr := installSetupPackages(files, dir); setupErr != nil {
-		log.Fatalf("Failed to install required packages: %v", setupErr)
-	}
+		// skip if install already complete
+		_, err = os.Stat(installCompleteFile)
+		if err == nil {
+			return
+		}
+
+		// lock to guard from concurrent artifact retrieval and installation,
+		// when called by child processes in a worker pool
+		lock, err := lockfile.New(filepath.Join(os.TempDir(), "beam.install.lck"))
```

Comment:
Should we separate `filepath.Join(os.TempDir(), "beam.install.lck")` to a variable similar to `installCompleteFile`? Not sure what is the go convention if it is being used only in one place.

Reply:
If it's only used in one place, this is fine as is. It would not be wrong to move the path definition elsewhere, but it would be indirect.